### DISCUSSION TITLE
Exception thrown when trying to retrieve assets from prefix in S3 bucket

### DISCUSSION THREAD

**Discussion by AndrewReid-Journey** [statamic/cms#7455] [Troubleshooting] [answered]
I'm working on a site which uses S3 as its asset storage and for the most part this works fine. I wanted to move the items out of the bucket root and move them into a prefix inside the bucket, however after copying the files to a prefix of 'myprefix' and updating the .env value to include this, the Statamic site throws the following exception when navigating to it. 

![image](https://user-images.githubusercontent.com/91478458/216303813-24de3695-e929-4cc1-a660-d341345298b4.png)

The .env values look like this : 

```
AWS_DEFAULT_REGION="eu-west-1"
AWS_BUCKET="testbucket-alpha/myprefix"
AWS_URL="https://testbucket-alpha.s3.eu-west-1.amazonaws.com/myprefix"
AWS_ENDPOINT="https://s3.eu-west-1.amazonaws.com"
AWS_USE_PATH_STYLE_ENDPOINT=true
```

I noticed in the exception as part of the GET url, there is a prefix argument that is blank which I imagine is part of the problem here. If that is the case, how can I get that prefix argument to update?

**Comment by duncanmcclean** [comment, marked as answer]
Try removing the `/myprefix` part of the `AWS_URL`.

Instead, in your `config/filesystems.php` file for the S3 disk, add the `root` option:

```php
'root' => 'myprefix',
```

**Reply by AndrewReid-Journey** [reply]
Amazing, thank you for that. I had to also remove the prefix from the `AWS_BUCKET` but works like a charm. Thanks again.
